### PR TITLE
ABI: Clarify padding direction.

### PR DESCRIPTION
**Pull request by chriseth**

### CODE REVIEW

File: docs/abi-spec.rst [hunk 1]
```diff
@@ -155,15 +155,15 @@ on the type of ``X`` being
...
 - ``address``: as in the ``uint160`` case
-- ``int<M>``: ``enc(X)`` is the big-endian two's complement encoding of ``X``, padded on the higher-order (left) side with ``0xff`` for negative ``X`` and with zero bytes for positive ``X`` such that the length is a multiple of 32 bytes.
+- ``int<M>``: ``enc(X)`` is the big-endian two's complement encoding of ``X``, padded on the higher-order (left) side with ``0xff`` for negative ``X`` and with zero bytes for positive ``X`` such that the length is 32 bytes.
 - ``bool``: as in the ``uint8`` case, where ``1`` is used for ``true`` and ``0`` for ``false``
 - ``fixed<M>x<N>``: ``enc(X)`` is ``enc(X * 10**N)`` where ``X * 10**N`` is interpreted as a ``int256``.
 - ``fixed``: as in the ``fixed128x19`` case
 - ``ufixed<M>x<N>``: ``enc(X)`` is ``enc(X * 10**N)`` where ``X * 10**N`` is interpreted as a ``uint256``.
 - ``ufixed``: as in the ``ufixed128x19`` case
-- ``bytes<M>``: ``enc(X)`` is the sequence of bytes in ``X`` padded with zero-bytes to a length of 32.
+- ``bytes<M>``: ``enc(X)`` is the sequence of bytes in ``X`` padded with zero-bytes on the _right_ to a length of 32.
```
Comment (axic) [inline docs/abi-spec.rst]: Do we want to have the same style here as for `int`?

> padded on the lower-order (right) side with zero-bytes such that the length is 32 bytes.

Comment (chriseth) [reply docs/abi-spec.rst]: There is no `lower-order` for bytes, they are not numbers.

Comment (chriseth) [reply docs/abi-spec.rst]: We need to use "lower-order" for numbers to clarify endianness.

Comment (axic) [reply docs/abi-spec.rst]: "to the right" doesn't sound correct for "bytes" in that case. "postfixing with zeroes" or "pad with trailing zeroes" might be clearer?

Comment (chriseth) [reply docs/abi-spec.rst]: ok!

**Pull request merged**